documentation/asciidoc/computers/raspberry-pi/boot-http.adoc

```diff
@@ -0,0 +1,113 @@
+
+When net install is enabled it detects a keyboard and displays the Net Install screen on boot. You can switch this behaviour on and off with this property.
+
+To enable network install add, `NET_INSTALL_ENABLED=1`, or to disable network install add `NET_INSTALL_ENABLED=0`.
+
+This setting is ignored and net install / keyboard detection on boot is disabled if `DISABLE_HDMI=1` is set.
+
+==== NET_INSTALL_KEYBOARD_WAIT
+
+If net install is enabled, the bootloader waits to detect a keyboard and the `SHIFT` key to initiate network install. You can change the length of this wait in milliseconds with this property,
+
+Default: `900`
+
+Adding `NET_INSTALL_KEYBOARD_WAIT=0` disables the keyboard wait, although network install can still be initiated if no boot files are found and USB boot `4` is in `BOOT_ORDER`.
+
+NOTE: Testing suggests Keyboard / shift detection takes at least 750ms.
+
+==== HTTP_HOST
+
+If a HTTP download is initiated boot.img and boot.sig are downloaded from this server.
+
+Default: `fw-download-alias1.raspberrypi.com`
+
+Invalid host names will be ignored.
+If `HTTP_HOST` is set then HTTPS is disabled and plain HTTP used instead.
+You should be able to specify an IP address and avoid the need for a DNS lookup.
+Don`t include the HTTP scheme or any forward slashes in the hostname.
+
+==== HTTP_PORT
+
+By default HTTP uses port `443` because HTTPS is enabled when using the Raspberry Pi host. If `HTTP_HOST` is changed then HTTPS is not supported and HTTP will be used instead, so port `80` becomes the default.
+
+You can override this and specify a different port, but this is ignored if `HTTP_HOST` is not set.
+
+==== HTTP_PATH
+
+You can change the path used to download files with this property.
+
+Default: `net_install`
+
+The case of the path *is* significant.
+Use forward (Linux) slashes for the path separator.
+Leading and trailing forward slashes are not required.
+Any value for `HTTP_PATH` will only be used if `HTTP_HOST` is also set.
+
+==== IMAGER_REPO_URL
+
+The imager application is configured with a json file which it downloads on start up.
+
+Default: http://downloads.raspberrypi.org/os_list_imagingutility_v3.json
+
+You can change the URL and json file used by the imager application to get it to supply your own images.
+Test this with the Windows or Linux Imager applications by passing the URL via the `--repo` argument.
+
+=== Keys
+
+All HTTP downloads must be signed. The Raspberry Pi public key will be used to verify the network install image unless `HTTP_HOST` is set and a custom public key is included in the eeprom.
+
+Adding `SIGNED_BOOT=0` to the eeprom configuration has no effect on HTTP downloads as they must always be signed.
+
+WARNING: Using your own net install image will require you to add your public key to the eeprom. At the moment, if you then apply a public eeprom update, your key will be lost and need to be re-added.
```

Comment:
side-question: Is there a plan for rpi-eeprom-update to "migrate" public keys during an upgrade, the same way as it currently "migrates" config values?

Reply:
yes - we have a cunning plan. It's just not done yet. I don't expect any people will use this feature yet.